embeddings/pipeline/lightning_pipeline.py

```diff
@@ -13,16 +19,56 @@ class LightningPipeline(
     Pipeline[EvaluationResult],
     Generic[Data, ModelResult, EvaluationResult],
 ):
+    DEFAULT_TASK_TRAIN_KWARGS: Dict[str, Any] = {"devices": "auto", "accelerator": "auto"}
+    DEFAULT_TASK_MODEL_KWARGS: Dict[str, Any] = {"use_scheduler": True}
+    DEFAULT_DATAMODULE_KWARGS: Dict[str, Any] = {"max_seq_length": None}
+    DEFAULT_MODEL_CONFIG_KWARGS: Dict[str, Any] = {"classifier_dropout": None}
+    DEFAULT_LOGGING_KWARGS: Dict[str, Any] = {
+        "use_tensorboard": True,
+        "use_wandb": True,
+        "use_csv": True,
+    }
+    DEFAULT_EARLY_STOPPING_KWARGS: Dict[str, Any] = {
+        "monitor": "val/Loss",
+        "mode": "min",
+        "patience": 3,
+    }
+
     def __init__(
         self,
         datamodule: BaseDataModule[Data],
         model: Model[BaseDataModule[Data], ModelResult],
         evaluator: Evaluator[ModelResult, EvaluationResult],
+        output_path: Path,
+        **kwargs: Any
     ) -> None:
         self.datamodule = datamodule
         self.model = model
         self.evaluator = evaluator
+        self.output_path = output_path
+
+    def run(self, run_name: Optional[str] = None) -> EvaluationResult:
+        if run_name:
+            run_name = standardize_name(run_name)
+        self._save_artifacts()
+        model_result = self.model.execute(data=self.datamodule, run_name=run_name)
+        result = self.evaluator.evaluate(model_result)
+        self._finish_logging()
+        return result
+
+    @property
+    @abstractmethod
+    def logging_kwargs(self) -> Dict[str, Any]:
+        pass
+
+    def _save_artifacts(self) -> None:
+        srsly.write_json(self.output_path.joinpath("packages.json"), get_installed_packages())
 
-    def run(self) -> EvaluationResult:
-        model_result = self.model.execute(data=self.datamodule)
-        return self.evaluator.evaluate(model_result)
+    def _finish_logging(self, run_name: Optional[str] = None) -> None:
+        if self.logging_kwargs["use_wandb"]:
+            wandb.log_artifact(
```

Comment:
Could we parametrize it as a wrapper? like we have some wrapper logic 
```python
class ExperimentsLogger:
    def log_artifacts(...):
           pass

class WandbLogger:
    def log_artifacts(...):
           wandb.log_artifcats(...)
```

How long would it take to apply such approach?

Reply:
wandb logging is already encapsulated in pytorch lightning, unfortunately it does not support all of the functionalities. Creating wrapper for only one function is complicating things unnecessarily. Anyway I added LightningLoggingConfig to wrap config

Reply:
Yes but it would speedup the process to add another logger.